src/main/java/com/palantir/gradle/versions/CheckBadPinsTask.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.versions;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.palantir.gradle.versions.FuzzyPatternResolver.Glob;
+import com.palantir.gradle.versions.lockstate.Line;
+import com.palantir.gradle.versions.lockstate.LockState;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.language.base.plugins.LifecycleBasePlugin;
+
+public class CheckBadPinsTask extends DefaultTask {
+
+    private final Property<Boolean> shouldFix = getProject().getObjects().property(Boolean.class);
+    private final RegularFileProperty propsFileProperty =
+            getProject().getObjects().fileProperty();
+    private final RegularFileProperty lockFileProperty =
+            getProject().getObjects().fileProperty();
+
+    public CheckBadPinsTask() {
+        shouldFix.set(false);
+        setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
+        setDescription(
+                "Ensures matched versions in your versions.lock are pinned to avoid wasted dependency resolution.");
+        getOutputs().upToDateWhen(_task -> true); // task has no outputs, this is needed for it to be up to date
+    }
+
+    final void setLockFile(File lockFile) {
+        this.lockFileProperty.set(lockFile);
+    }
+
+    final void setPropsFile(File propsFile) {
+        this.propsFileProperty.set(propsFile);
+    }
+
+    @InputFile
+    public final Property<RegularFile> getLockFile() {
+        return lockFileProperty;
+    }
+
+    @InputFile
```

Comment:
When combined in the abstract version, this should be `@OutputFile` too

Reply:
Actually, I don't think gradle likes this, the `outputs` thing in the constructor is the correct thing.